targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/PinNames.h

```diff
@@ -0,0 +1,241 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_PINNAMES_H
+#define MBED_PINNAMES_H
+
+#include "cmsis.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+        PIN_INPUT,
+        PIN_OUTPUT
+} PinDirection;
+
+#define PORT_SHIFT  5
+
+typedef enum {
+    // MPS2 EXP Pin Names
+        EXP0 = 0 ,
+        EXP1 = 4 ,
+        EXP2 = 2 ,
+        EXP3 = 3 ,
+        EXP4 = 1 ,
+        EXP5 = 15,
+        EXP6 = 5 ,
+        EXP7 = 6 ,
+        EXP8 = 7 ,
+        EXP9 = 8 ,
+        EXP10 =9 ,
+        EXP11 =13,
+        EXP12 =10,
+        EXP13 =11,
+        EXP14 =12,
+        EXP15 =14,
+        EXP16 =18,
+        EXP17 =19,
+        EXP18 =20,
+        EXP19 =21,
+        EXP20 =52,
+        EXP21 =53,
+        EXP22 =54,
+        EXP23 =55,
+        EXP24 =56,
+        EXP25 =57,
+
+        EXP26 =16,
+        EXP27 =25,
+        EXP28 =24,
+        EXP29 =31,
+        EXP30 =17,
+        EXP31 =23,
+        EXP32 =27,
+        EXP33 =30,
+        EXP34 =26,
+        EXP35 =28,
+        EXP36 =29,
+        EXP37 =58,
+        EXP38 =48,
+        EXP39 =49,
+        EXP40 =50,
+        EXP41 =22,
+        EXP42 =59,
+        EXP43 =60,
+        EXP44 =51,
+        EXP45 =61,
+        EXP46 =62,
+        EXP47 =63,
+        EXP48 =64,
+        EXP49 =65,
+        EXP50 =66,
+        EXP51 =67,
+
+// Other mbed Pin Names
+
+        //LEDs on mps2
+        //user leds
+        USERLED1 = 100,
+        USERLED2 = 101,
+        //user switches
+        USERSW1  = 110,
+        USERSW2  = 111,
+
+        //mcc leds
+        LED1 = 200,
+        LED2 = 201,
+        LED3 = 202,
+        LED4 = 203,
+        LED5 = 204,
+        LED6 = 205,
+        LED7 = 206,
+        LED8 = 207,
+
+        //MCC Switches
+        SW1 = 210,
+        SW2 = 211,
+        SW3 = 212,
+        SW4 = 213,
+        SW5 = 214,
+        SW6 = 215,
+        SW7 = 216,
+        SW8 = 217,
+
+        //MPS2 SPI header pins j21
+        MOSI_SPI = 300,
+        MISO_SPI = 301,
+        SCLK_SPI = 302,
+        SSEL_SPI = 303,
```

Comment:
We don't have a hard standard for this, however if you look through the code base, the majority of targets use the naming convention `SPI_X` instead of `X_SPI`. For example, `SPI_MOSI` instead of `MOSI_SPI`. This is not required, however you will be compatible with more example programs in the mbed ecosystem if you follow this convention.